tests/unit/test_filehandler.py

```diff
@@ -155,7 +154,144 @@ def test_baserotatinghandler_callable_namer(tmp_path):
     logger.warning("test")
     assert sorted(os.listdir(tmp_path)) == ["log.txt", "log.txt.1.5"]
 
+
 def test_filehandler_repr(tmp_path):
     log_file = tmp_path / "log.txt"
     handler = picologging.FileHandler(log_file)
     assert repr(handler) == f"<FileHandler {log_file} (NOTSET)>"
+
+
+@pytest.mark.parametrize("utc", [False, True])
+def test_timed_rotatingfilehandler_rollover(tmp_path, utc):
+    log_file = tmp_path / "log.txt"
+    handler = TimedRotatingFileHandler(log_file, when="S", backupCount=2, utc=utc)
+    logger = picologging.getLogger("test")
+    logger.setLevel(picologging.DEBUG)
+    logger.addHandler(handler)
+
+    logger.warning("test")
+    handler.rollover_at = time.time() - 1
+
+    logger.warning("test")
+    for file_name in os.listdir(tmp_path):
+        with open(tmp_path / file_name, "r") as file:
+            assert file.read() == "test\n"
+
+
+def test_timed_rotatingfilehandler_rollover_removes_old_files(tmp_path):
+    with open(tmp_path / "log.txt.1970-11-01_00-00-00", "w"):
+        ...
+
+    log_file = tmp_path / "log.txt"
+    handler = TimedRotatingFileHandler(log_file, when="S", backupCount=1)
+    logger = picologging.getLogger("test")
+    logger.setLevel(picologging.DEBUG)
+    logger.addHandler(handler)
+
+    logger.warning("test")
+    handler.rollover_at = time.time() - 1
+    logger.warning("test")
+    
+    assert len(os.listdir(tmp_path)) == 2
```

Comment:
The rollover should remove the existing files if they match the log format to keep backupCount.